src/tickit/core/management/schedulers/base.py

```diff
@@ -35,6 +44,8 @@ def __init__(
         self._state_producer_cls = state_producer
         self.wakeups: Dict[ComponentID, SimTime] = dict()
 
+        self.error: asyncio.Event = asyncio.Event()
```

Comment:
This type can be trivially inferred, therefore we can elide the type hint 
```suggestion
        self.error = asyncio.Event()
```